sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeExtractors.scala

```diff
@@ -394,7 +394,7 @@ trait GetMapValueUtil extends BinaryExpression with ImplicitCastInputTypes {
     val keyJavaType = CodeGenerator.javaType(keyType)
     nullSafeCodeGen(ctx, ev, (eval1, eval2) => {
       val keyNotFoundBranch = if (failOnError) {
-        s"""throw new NoSuchElementException("Key " + $eval2 + " does not exist.");"""
+        s"""throw new java.util.NoSuchElementException("Key " + $eval2 + " does not exist.");"""
```

Comment:
@leanken, can we add a test?

Reply:
> @leanken, can we add a test?

Currently, this case being missed in SPARK-33460 it is because checkExceptionInExpression is not as expected. And as i investigated, there are some cases would fail if I correct the behavior of checkExceptionInExpression by using withSQLConf set CodeGenFallBack to false, like ObjectExpressionsSuite and RegexpExpressionsSuite and ETC.

So I need your advise. Should I
1. Fix all these wrong behavior separately and then final fix checkExceptionInExpression.
2. Fix all these wrong behavior and checkExceptionInExpression with the same patch.

The reason why i submit this PR it's because this bug introduced by me and I know of the history, but all those other behavior is fresh to me, and the all-in-one fix might be big.

As for add a test to the java.util.NoSuchElementException, I think the offline manual verification is enough, If we fix checkExceptionInExpression, and make it reliable, then we don't need to check code gen behavior for every new added or updated operation, right?

Reply:
> @leanken, can we add a test?

maybe you can try repo ObjectExpressionsSuite and RegexpExpressionsSuite in your environment so that you can get clearer picture.

Reply:
> And as i investigated, there are some cases would fail if I correct the behavior of checkExceptionInExpression by using withSQLConf set CodeGenFallBack to false, like ObjectExpressionsSuite and RegexpExpressionsSuite and etc.

How about making a PR to fix `checkExceptionInExpression` first? I'm currently not sure how many failures caused by it exist now. If there are too many failures, we can separate work to fix them then.

Reply:
+1 for @maropu 's advice.